synapse/handlers/room_member.py

```diff
@@ -416,6 +417,10 @@ def _update_membership(
                 ret = yield self._remote_join(
                     requester, remote_room_hosts, room_id, target, content
                 )
+                logcontext.run_in_background(
```

Comment:
does this really need to run in the background? can't we block for it?

Reply:
There's the potential for tens, possibly even hundreds, of invites to be sent. It's not something critical we need to block on either.